core/consensus/babe/impl/block_executor.hpp

```diff
@@ -97,14 +97,49 @@ namespace kagome::consensus {
     std::shared_ptr<primitives::BabeConfiguration> genesis_configuration_;
     std::shared_ptr<BabeSynchronizer> babe_synchronizer_;
     std::shared_ptr<BlockValidator> block_validator_;
-    std::shared_ptr<consensus::JustificationValidator> justification_validator_;
+    std::shared_ptr<grandpa::Environment> grandpa_environment_;
     std::shared_ptr<EpochStorage> epoch_storage_;
     std::shared_ptr<transaction_pool::TransactionPool> tx_pool_;
     std::shared_ptr<crypto::Hasher> hasher_;
     std::shared_ptr<authority::AuthorityUpdateObserver>
         authority_update_observer_;
     const SlotsStrategy slots_strategy_;
+    std::shared_ptr<boost::asio::io_context> io_context_;
     common::Logger logger_;
+
+    class AsyncHelper : public std::enable_shared_from_this<AsyncHelper> {
+     public:
+      AsyncHelper() = delete;                         // Default-constructor
+      AsyncHelper(AsyncHelper &&) noexcept = delete;  // Move-constructor
+      AsyncHelper(const AsyncHelper &) = delete;      // Copy-constructor
+
+      AsyncHelper(std::shared_ptr<boost::asio::io_context> context)
+          : std::enable_shared_from_this<AsyncHelper>(),
+            io_context_(std::move(context)){};
+
+      ~AsyncHelper() = default;  // Destructor
```

Comment:
I have slight doubts about the necessity of this comment.

Reply:
ok